version.go

```diff
@@ -105,19 +106,20 @@ const (
 	// - 1.2.3.4
 	// - 1.2-alpha3
 	// - 1.2-alpha3.4
-	NumberRegex = `(\d{1,9})\.(\d{1,9})(?:\.|-([a-z]+))(\d{1,9})(\.\d{1,9})?`
+	NumberRegex = `(?P<major>\d{1,9})(\.((?P<minor>\d{1,9})((?:(-((?P<tag>[a-z]+)(?P<patchInTag>\d{1,9})?)?)|(\.(?P<patch>\d{1,9})))?)(\.(?P<build>\d{1,9}))?)?)?`
```

Comment:
offhand i feel like there are too many "?" at the end. If I am parsing through it correctly you support
```
1
1. <- I don't think we want to support this one
1.2
1.2- <- also don't want this one
```

Interestingly semver doesn't support 1.2-tag, but does allow 1.2.3-tag (which *we* support in the opposite way)

I played around on ihateregex giving it some of those patterns and seeing what resulted. I think you want to remove 2 "?" from your regex:
```^(\d{1,9})(\.((\d{1,9})((?:(-(([a-z]+)(\d{1,9})?))|(\.(\d{1,9})))?)(\.(\d{1,9}))?))?$```
but that also removes some other things. Lets see if I can get it right:
```suggestion
	NumberRegex = `(?P<major>\d{1,9})(\.((?P<minor>\d{1,9})((?:(-((?P<tag>[a-z]+)(?P<patchInTag>\d{1,9})?))|(\.(?P<patch>\d{1,9})))?)(\.(?P<build>\d{1,9}))?))?`
```

Reply:
They shouldn't be matched; I will add some extra tests to be sure.

Reply:
Actually the `1.` did get matched; need to fix that

Reply:
Yes, I need to play around with it a bit more in debuggex.com to clean up the optional bits a bit more